share/docker.sh

```diff
@@ -158,12 +161,10 @@ docker_build()
         docker history "${image}" > "${histfile}"
         docker inspect "${image}" > "${inspfile}"
 
-        opt_forward docker_pull image registry username password cache_from -- ${tag[*]:-}
+        opt_forward docker_tag image -- ${tag[*]:-}
```

Comment:
This code was never meant to call `docker_pull` here. It was convenient as that function also created the tags we needed. So I broke that out into a new `docker_tag` function